src/RecoveryServices/RecoveryServices/Common/PSRecoveryServicesVaultClient.cs

```diff
@@ -111,9 +112,9 @@ public List<Microsoft.Azure.Management.Internal.Resources.Models.ResourceGroup>
         /// <param name="resouceGroupName">Name of the resouce group</param>  
         /// <param name="vaultName">Name of the vault</param>  
         /// <returns>Azure Resource Storage response object.</returns>  
-        public BackupStorageConfig GetVaultStorageType(string resouceGroupName, string vaultName)
+        public BackupResourceConfigResource GetVaultStorageType(string resouceGroupName, string vaultName)
```

Comment:
Please try to use BackupResourceConfig as the replacement for BackupStorageConfig being used earlier

Reply:
This method GetRecoveryServicesBackupClient.BackupResourceStorageConfigs.GetWithHttpMessagesAsync(
                resouceGroupName, vaultName, GetRequestHeaders()).Result.Body; returns BackupResourceConfigResource which contains BackupResourceConfig as properties field which we are using it.

Reply:
Okay, your changes look fine